### PR TITLE
add-reducers-actionsPullRequest

### LINKED ISSUE #3: add-reducers-actions
### Issue: add-reducers-actions
This part of the project is about understanding the concept of state management and using store, actions, and reducers
in React.

For it : 
- I created a featured branch named addreducers-actions.
- I will implement Redux Toolkit inside my react application.
- I will setup a Redux Store creating a folder named redux.
- Inside the redux folder I will store a file named store.js
- Then, I will create a slice of state for books to save state and reducers.
- Also, I will create a slice of state for categories
- I will pass linters ESLint and Stylelint tools.

### Goal
The goal is to understand well the concepts of state management and using store, actions, and reducers in React.

### PR DESCRIPTION
**Pull request by IvonneBenitesRodriguez**
### About this project:
This part of the project is about understanding the usage of Redux and its concept of state management 
and using store, actions, and reducers in React.

⭐️ For it :
 - I created a featured branch named addreducers-actions.
- I will implement Redux Toolkit inside my react application.
- I will setup a Redux Store creating a folder named redux.
- Inside the redux folder I will store a file named store.js
- Then, I will create a slice of state for books to save state and reducers.
- Also, I will create a slice of state for categories
- I will pass linters ESLint and Stylelint tools.

### What is the part of this project that you consider important ?
According my opinion the creation of a slice to save status and reducers as well as creating the store,  and provide store part are importants to understand the usage of Redux.

### CODE REVIEW

File: README.md
Comment (Efiamotu-1) [inline README.md:1]: - [ ] The authors section of your readme file currently contains only texts rather they should be links to your `GItHub, Twitter & linkedIn` accounts please kinldy add them appropriately :+1: :sparkles:

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = 'Under construction';
```
Comment (Efiamotu-1) [inline src/redux/categories/categoriesSlice.js]: - [ ] The inital state of your categories reducer should be an empty array. see tthe code snippet below as a guide
- code snippet
```javascript
 initialState: {
    categories: [],
  },
```

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => state,
```
Comment (Efiamotu-1) [inline src/redux/categories/categoriesSlice.js]: - [ ] The checkStatus action should receive a payload which checks if the categories is under construction, if not it should return the categories state. see the code snippet below as a guide 
- code snippet
```javascript
checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    }
```

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @IvonneBenitesRodriguez , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md
Comment (IvonneBenitesRodriguez) [reply README.md:1]: Dear @Efiamotu-1 👋🏾
Very nice to meet you,
thanks for the suggestion, I appreciate it,
I already modified this part.
Best regards,

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (IvonneBenitesRodriguez) [reply src/redux/categories/categoriesSlice.js]: Dear @Efiamotu-1 👋🏾
Thanks for the suggestion, and feedback,
 I appreciate it,
I already modified this part.
Best regards,

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (IvonneBenitesRodriguez) [reply src/redux/categories/categoriesSlice.js]: Dear @Efiamotu-1 👋🏾
Thanks for the suggestion, and feedback,
I appreciate it,
I already modified this part.
Best regards,

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @IvonneBenitesRodriguez 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- Used  ducks pattern structure for your files :heavy_check_mark: 
- Used `@reduxjs/toolkit` & `react-redux` :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**